src/dataframe/dataframe.jl

```diff
@@ -1090,14 +1090,60 @@ Base.convert(::Type{DataFrame}, A::AbstractMatrix) = DataFrame(A)
 
 Base.convert(::Type{DataFrame}, d::AbstractDict) = DataFrame(d)
 
+"""
+    push!(df1::DataFrame, row::Union{AbstractDict, NamedTuple})
 
-##############################################################################
-##
-## push! a row onto a DataFrame
-##
-##############################################################################
+Add in-place one row at the end of `df1` taking the values from `row`.
+
+Adding values is based on column name matching. `row` may contain more columns
+than `df1`, but all column names that are present in `df1` must be present in `row`.
+
+Column types of `df1` are preserved, and new values are converted if necessary.
+An error is thrown if conversion fails.
+
+If `df1` has no columns then it is allowed to `push!` a `NamedTuple` to `df1`
+and in this case all fields from `NamedTuple` are added for `df1` as a first row.
+
+# Examples
+```jldoctest
+julia> df = DataFrame(A=1:3, B=1:3)
+3×2 DataFrame
+│ Row │ A     │ B     │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 1     │ 1     │
+│ 2   │ 2     │ 2     │
+│ 3   │ 3     │ 3     │
 
+julia> push!(df, (C="something", A=true, B=false))
```

Comment:
I feel like we should deprecate this behavior and require setting an argument. What do you think?

Reply:
OK